clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/NodeStateChangeChecker.java

```diff
@@ -268,6 +284,60 @@ private Result anotherNodeInAnotherGroupHasWantedState(StorageNodeInfo nodeInfo)
         }
     }
 
+    /**
+     * Returns an optional Result, where return value is:
+     * For flat setup: Return Optional.of(disallowed) if wanted state is set on some node, else Optional.empty
+     * For hierarchical setup: No wanted state for other nodes, return Optional.empty
+     *                         Wanted state for nodes/groups are not UP:
+     *                           if less than maxNumberOfGroupsAllowedToBeDown: return Optional.of(allowed)
+     *                           else: if node is in group with nodes already down: return Optional.of(allowed), else Optional.of(disallowed)
+     */
+    private Optional<Result> otherNodesHaveWantedState(StorageNodeInfo nodeInfo, String newDescription) {
+        Node node = nodeInfo.getNode();
+
+        if (groupVisiting.isHierarchical()) {
+            if (maxNumberOfGroupsAllowedToBeDown <= 1) {
```

Comment:
Remove this `if` branch?  Since `maxNumberOfGroupsAllowedToBeDown == -1` is handled elsewhere and serve as the old behavior, we can handle `maxNumberOfGroupsAllowedToBeDown == 1` in the same way as `maxNumberOfGroupsAllowedToBeDown > 1` and by this `else` branch.